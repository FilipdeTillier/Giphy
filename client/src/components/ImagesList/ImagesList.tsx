import React from 'react';

interface ImagesListProps {
  images: string[];
}

const ImagesList: React.FC<ImagesListProps> = ({ images }) => {
  return (
    <>
      <div className="images-list">
        {images.map((image) => (
          <div key={image} className="card">
            <img
              className="card__image"
              src={image}
              alt={image}
              width="100%"
              height="100%"
            ></img>
          </div>
        ))}
      </div>
    </>
  );
};

export default ImagesList;
