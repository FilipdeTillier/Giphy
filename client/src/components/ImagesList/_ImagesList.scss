.images-list {
  max-width: 80%;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
  .card {
    width: 100%;
    margin-bottom: 40px;
    &__image {
      width: 100%;
      height: 100%;
    }
  }
}

@media all and (min-width: $md-screen) {
  .images-list {
    justify-content: space-between;
    flex-direction: row;
    .card {
      width: 45%;
    }
  }
}

@media all and (min-width: $lg-screen) {
  .images-list {
    .card {
      width: 30%;
    }
  }
}
