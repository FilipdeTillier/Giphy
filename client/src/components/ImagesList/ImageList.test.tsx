import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ImagesList from './ImagesList';

const images = new Array(10)
  .fill('')
  .map((el: string, index: number) => index.toString());

const loadMore = jest.fn();

const setup = () => {
  const container = render(<ImagesList images={images} loadMore={loadMore} />);
  const renderedImages: HTMLElement[] = container.getAllByRole('img');
  const loadMoreButton: HTMLElement = container.getByTestId('loadMore');
  return {
    container,
    renderedImages,
    loadMoreButton,
  };
};

test('render component proprely', () => {
  const { container } = setup();
  expect(container).toMatchSnapshot();
});

test('should render 10 elements and load button', () => {
  const { renderedImages, loadMoreButton } = setup();
  expect(renderedImages).toHaveLength(10);
  expect(loadMoreButton).toBeTruthy();
});

test('click button should run method', () => {
  const { loadMoreButton } = setup();
  fireEvent.click(loadMoreButton);
  expect(loadMore).toHaveBeenCalledTimes(1);
});
