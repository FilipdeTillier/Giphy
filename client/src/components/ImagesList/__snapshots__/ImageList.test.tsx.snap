// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component proprely 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="images-list"
      >
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="0"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="1"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="2"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="3"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="4"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="5"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="6"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="7"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="8"
            width="100%"
          />
        </div>
        <div
          class="card"
        >
          <img
            class="card__image"
            height="100%"
            src="9"
            width="100%"
          />
        </div>
      </div>
      <div
        class="images-button"
      >
        <button
          data-testid="loadMore"
          type="button"
        >
          Load more
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="images-list"
    >
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="0"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="1"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="2"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="3"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="4"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="5"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="6"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="7"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="8"
          width="100%"
        />
      </div>
      <div
        class="card"
      >
        <img
          class="card__image"
          height="100%"
          src="9"
          width="100%"
        />
      </div>
    </div>
    <div
      class="images-button"
    >
      <button
        data-testid="loadMore"
        type="button"
      >
        Load more
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
